### ISSUE TITLE
bug:  Full ImportFromJson: DropTables Foreign Key constraint failed

### ISSUE THREAD

**Issue by tobiasmuecksch** [capacitor-community/sqlite#245]
# Description

Let's say I start the app for the first time and create the database `products-db` with the following json.

Afterwards I restart the app (without changing the version number).
For some reason the `importFromJson` method ignores that the version is the same and try to replace the database (and the error below occurs).

Then, I increment the version number in the json by one and in the `openDB` method call and restart the app. And the error occurs also.

## Error message

```
 ImportFromJson: ImportJson: CreateDatabaseSchema: DropAll: DropElements: DropTables: run: FOREIGN KEY constraint failed
```

## JSON Data structure

```json
{
  "database": "product-db",
  "version": 1,
  "encrypted": false,
  "mode": "full",
  "tables": [
    {
      "name": "vendors",
      "schema": [
        { "column": "id", "value": "INTEGER PRIMARY KEY NOT NULL" },
        { "column": "company_name", "value": "TEXT NOT NULL" },
        { "column": "company_info", "value": "TEXT NOT NULL" },
        { "column": "last_modified", "value": "INTEGER DEFAULT (strftime('%s', 'now'))" }
      ],
      "values": [
        [1, "Devdactic", "The main blog of Simon Grimm", 1587310030],
        [2, "Ionic Academy", "The online school to learn Ionic", 1590388125],
        [3, "Ionic Company", "Your favourite cross platform framework", 1590383895]
      ]
    },
    {
      "name": "products",
      "schema": [
        { "column": "id", "value": "INTEGER PRIMARY KEY NOT NULL" },
        { "column": "name", "value": "TEXT NOT NULL" },
        { "column": "currency", "value": "TEXT" },
        { "column": "value", "value": "INTEGER" },
        { "column": "vendorid", "value": "INTEGER" },
        { "column": "last_modified", "value": "INTEGER DEFAULT (strftime('%s', 'now'))" },
        {
          "foreignkey": "vendorid",
          "value": "REFERENCES vendors(id)"
        }
      ],
      "values": [
        [1, "Devdactic Fan Hat", "EUR", 9, 1, 1604396241],
        [2, "Ionic Academy Membership", "USD", 25, 2, 1604296241],
        [3, "Ionic Sticker Swag", "USD", 4, 3, 1594196241],
        [4, "Practical Ionic Book", "USD", 79, 1, 1603396241]
      ]
    }
  ]
}
```

## Conclusion
I therefore see two bugs:

1. The import method should check the version, and if it is equal it should either silently do nothing or throw an error like "database already is on version 1"
2. The Table Drop procedure should ignore foreign key constraints. Otherwise it would have to delete the tables in a specific order.

**Additional context**
I'm testing my code in safari. If you need some reproduction demo, just let me know. I'll provide you.

**Comment by jepiqueau** [comment]
@tobiasmuecksch thanks for your feedback. For the drop table, i thought i did put the foreign key constraints off but i should lost it in between two releases. for the import you right i am not checking the version, so i have to do it. i do not think it is a good idea to throw an error as when you test your app you will have to increase the version number each time which for me is stupid. the version of the database change when the database schema changes.

**Comment by jepiqueau** [comment]
@tobiasmuecksch i am still working in.

**Comment by tobiasmuecksch** [comment]
All good! Thank you for the effort you put into this plugin and your friendly support! 😃

**Comment by jepiqueau** [comment]
@tobiasmuecksch i updated the jeep-sqlite accordingly so if you can test the Web part and tell me if it works.

**Comment by jepiqueau** [comment]
@tobiasmuecksch fix in 3.4.2-3
🚨 Since release 3.4.2-3 ->> 🚨

 - **overwrite** boolean parameter has been added to the Json Object (default false) 
   - `true` : delete the physically the database whatever the version is.
   - `false`: 
      - re-importing a database with the same `version` number will do nothing, keeping the existing database and will return `changes = 0`
      - re-importing a database with a lower `version` number will throw an error `ImportFromJson: Cannot import a version lower than `

 - During an import in `full` mode the `Foreign Key` constraint has been turn off before dropping the tables and turn back on after

🚨 Since release 3.4.2-3 <<- 🚨

**Comment by tobiasmuecksch** [comment]
Seems to work as expected. Thank you for your awesome work!
